### PR TITLE
Add app: cavity flow(draft) 

### PR DESCRIPTION
**Pull request by EN-Chou**
This PR includes two apps: `cavity2d_noQt` and `cavity2d`. Both apps share the same code, but differ in their pipeline and status monitoring. `cavity2d_noQt` does not include Qt code in the pipeline, while `cavity2d` does. Additionally, `cavity2d_noQt` can be monitored through printed information in the terminal, while in `cavity2d`, status information is only visible in Modmesh Viewer when the app is halted with `<ctrl> + C` in the terminal.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,7 +32,7 @@ CMAKE_INSTALL_PREFIX ?= $(MODMESH_ROOT)/build/fakeinstall
 CMAKE_LIBRARY_OUTPUT_DIRECTORY ?= $(MODMESH_ROOT)/modmesh
 # Use CMAKE_PREFIX_PATH to make it easier to build with Qt, e.g.,
 # CMAKE_PREFIX_PATH=/path/to/qt/6.2.3/macos
-CMAKE_PREFIX_PATH ?=
+CMAKE_PREFIX_PATH ?=/usr/local/Qt-6.3.1/
```
Comment (yungyuc) [inline Makefile]: Please do not hard-code `CMAKE_PREFIX_PATH`.

File: cpp/modmesh/view/RManager.cpp [hunk 2]
```diff
@@ -184,6 +184,8 @@ void RManager::setUpMenu()
         this->addApplication(QString("euler1d"));
         this->addApplication(QString("linear_wave"));
         this->addApplication(QString("bad_euler1d"));
+        this->addApplication(QString("cavity2d_noQt"));
```
Comment (yungyuc) [inline cpp/modmesh/view/RManager.cpp]: `cavity2d` and `cavity2d_noQt` look similar and should be merged into one module/app.

File: modmesh/app/cavity2d.py [hunk 3]
```diff
@@ -0,0 +1,370 @@
+# flake8: noqa
```
Comment (yungyuc) [inline modmesh/app/cavity2d.py]: Do not turn off flake8 unconditionally.

**Review by yungyuc — CHANGES_REQUESTED** [review]
Please see the comments in line for improvement.

**Comment by EN-Chou** [comment]
Revision checklist:
- [x] edit CMAKE_PREFIX_PATH
- [x] merge `cavity2d_noQt` and `cavity2d`
- [x] revise the use of flake8

File: modmesh/app/cavity2d.py [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+
+
+ctrl = mm.app.cavity2d.run(interval=10, max_steps=50, profiling=True)
+""")
+    cmd = "ctrl = mm.app.cavity2d.run(interval=10, max_steps=50)"
+    view.mgr.pycon.command = cmd
+
+
+###############################################################################
+N = 81
```
Comment (yungyuc) [inline modmesh/app/cavity2d.py]: Please avoid using so many global variables.  Instead, create a class to hold all the values and pass it to the functions or classes that need them.

File: modmesh/app/cavity2d.py [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+V_2 = np.zeros((N + 1, N + 1))
+U_prev = np.zeros((N + 1, N + 1))
+V_prev = np.zeros((N + 1, N + 1))
+P_prev = np.zeros((N + 1, N + 1))
+UC = np.zeros((N, N))
+VC = np.zeros((N, N))
+PC = np.zeros((N, N))
+
+# benchmark
+x_b = [0, 0.0625, 0.0703, 0.0781, 0.0938, 0.1563, 0.2266, 0.2344, 0.5000, 0.8047, 0.8594, 0.9063, 0.9453, 0.9531, 0.9609, 0.9688, 1.0]  # noqa: E501
```
Comment (yungyuc) [inline modmesh/app/cavity2d.py]: You should not use lines that are too long.  Please shorten and remove skipping E501.

**Review by yungyuc — CHANGES_REQUESTED** [review]
I suggest to squash the multitude of the commits of this PR.

**Comment by EN-Chou** [comment]
@yungyuc thanks for your feedback. The revision checklist will include:
- [x] Shorten lines and remove skipping E501.
- [x] Create a class to hold all the values and pass it to the functions or classes.
- [ ] Squash commits in this PR.

**Comment by yungyuc** [comment]
Next time you update the PR please copy the tasks to the PR description body.  Github shows the number of tasks done and remaining.  It will be easier to track the progress.

**Comment by yungyuc** [comment]
@EN-Chou could you please comment at the code to summarize what you updated?  It has been a while.  Your summary helps a lot to the review.

**Comment by EN-Chou** [comment]
> @EN-Chou could you please comment at the code to summarize what you updated? It has been a while. Your summary helps a lot to the review.

I apologize for not mentioning earlier that the current status of the project is not yet ready for review. Last week, I made a mistake when using rebase and ended up force pushing, which caused some problems. However, these issues were resolved during the meetup yesterday. There are still a few remaining mistakes that need to be addressed, and I will provide a summary of these shortly.

**Comment by yungyuc** [comment]
I see.  It would help the community know the nature of the updates are not for review by making an explicit comment saying that.  Marking a PR work in progress does not stop people from seeing the updates, and we cannot tell whether or not the updater wants it to be reviewed or not.  Explicit is better than implicit.

**Comment by tigercosmos** [comment]
@EN-Chou do you want to continue the PR?

**Comment by yungyuc** [comment]
@EN-Chou if you don't respond by the end of this week, I will close this PR.

**Pull request closed without merging**